src/SqlConnectionConfig.ts

```diff
@@ -60,150 +64,123 @@ export default class SqlConnectionConfig {
      */
     private _maskSecrets(): void {
         // User ID could be client ID in some authentication types
-        if (this._connectionConfig.user) {
-            core.setSecret(this._connectionConfig.user);
+        if (this._parsedConnectionString.userId) {
+            core.setSecret(this._parsedConnectionString.userId);
         }
 
-        if (this._connectionConfig.password) {
-            core.setSecret(this._connectionConfig.password);
+        if (this._parsedConnectionString.password) {
+            core.setSecret(this._parsedConnectionString.password);
         }
     }
 
+    private _parseConnectionString(connectionString: string): SqlConnectionString {
```

Comment:
Can you look through https://github.com/tediousjs/connection-string to see if there's any logic that we're missing? Anywhere where we're parsing connection strings makes me a bit paranoid 😰.

Also curious if you took a look at leveraging that library as a dependency. Looks like it's pretty widely used: https://www.npmjs.com/package/@tediousjs/connection-string

Reply:
I did look into that library, it is what was used under the hood in v2 for parsing. I just switched to the old regex parsing logic from v1.

The parsing library has no external dependencies so we can take it if you want.

Reply:
I think I'd rather use that that have our homegrown connection string parsing, to be honest. Would you mind making the change, if it's not too much work?

Reply:
Updated to use that library instead